pkg/csi/cinder/controllerserver.go

```diff
@@ -136,7 +136,8 @@ func (cs *controllerServer) CreateVolume(ctx context.Context, req *csi.CreateVol
 		// In case a snapshot is not found
 		// check if a Backup with the same ID exists
 		if backupsAreEnabled && cpoerrors.IsNotFound(err) {
-			back, err := cloud.GetBackupByID(snapshotID)
+			var back *backups.Backup
+			back, err = cloud.GetBackupByID(snapshotID)
```

Comment:
without this change, restore will only create empty new pvc as result. With this change it will also restore the data for the volume